BBB_cci_mpf/sw/src/mpf.cmake

```diff
@@ -24,40 +24,41 @@
 ## ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE,  EVEN IF ADVISED OF THE
 ## POSSIBILITY OF SUCH DAMAGE.
 
+set(CMAKE_CXX_STANDARD 11)
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -pthread")
+
 file(
-    GLOB_RECURSE
+    GLOB
     HDR
-    FOLLOW_SYMLINKS
     ${PROJECT_SOURCE_DIR}/include/opae/mpf/*.h
     )
+file(
+    GLOB
+    HDR_CXX
+    ${PROJECT_SOURCE_DIR}/include/opae/mpf/cxx/*.h
+    )
 
 aux_source_directory(
-    ${PROJECT_SOURCE_DIR}/src
-    SRC
+    ${PROJECT_SOURCE_DIR}/src/libmpf
+    LIBMPF
+    )
+aux_source_directory(
+    ${PROJECT_SOURCE_DIR}/src/libmpf++
+    LIBMPF_CXX
     )
 
-add_library(MPF SHARED ${SRC})
+add_library(MPF SHARED ${LIBMPF})
+add_library(MPF-cxx SHARED ${LIBMPF_CXX})
 
 install(
-    TARGETS MPF
+    TARGETS MPF MPF-cxx
     RUNTIME DESTINATION bin
-    LIBRARY DESTINATION lib
+    LIBRARY DESTINATION lib64
```

Comment:
Is this only supported on 64-bit platforms?

Reply:
What's the right way to support both?

Reply:
Maybe check the global property `FIND_LIBRARY_USE_LIB64_PATHS`?
https://cmake.org/cmake/help/v3.0/prop_gbl/FIND_LIBRARY_USE_LIB64_PATHS.html

OPAE codebase uses this property to determine whether to install to lib or lib64:

https://github.com/OPAE/opae-sdk/blob/cfcfc259e43ef0171535421e29b3798fa99269d0/cmake/modules/fpga_functions.cmake#L79-L84